### PR TITLE
provider/aws: Add aws_default_route_table resource

### PR DESCRIPTION
**Pull request by catsby**
Adds a resource to manage the default route table that comes with every VPC. 
This is an advanced resource with caveats mentioned in the documentation. 

Part of #6093

### CODE REVIEW

File: builtin/providers/aws/resource_aws_default_route_table.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	}
+
+	rt := rtRaw.(*ec2.RouteTable)
+
+	d.Set("vpc_id", rt.VpcId)
+
+	// revoke all default and pre-existing routes on the default route table.
+	// In the UPDATE method, we'll apply only the rules in the configuration.
+	log.Printf("[DEBUG] Revoking default routes for Default Route Table for %s", d.Id())
+	if err := revokeAllRouteTableRules(d.Id(), meta); err != nil {
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_default_route_table.go]: Is there a potentially for this to cause issues if the revoke happens and the update fails?

File: builtin/providers/aws/resource_aws_default_route_table.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	filter2 := &ec2.Filter{
+		Name:   aws.String("vpc-id"),
+		Values: []*string{aws.String(d.Get("vpc_id").(string))},
+	}
+
+	findOpts := &ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{filter1, filter2},
+	}
+
+	resp, err := conn.DescribeRouteTables(findOpts)
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_default_route_table.go]: Would it not be best to use d.Id here as part of the lookup?

File: builtin/providers/aws/resource_aws_default_route_table.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	if err != nil {
+		return err
+	}
+
+	if len(resp.RouteTables) < 1 || resp.RouteTables[0] == nil {
+		return fmt.Errorf("Default Route table not found")
+	}
+
+	rt := resp.RouteTables[0]
```
Comment (stack72) [inline builtin/providers/aws/resource_aws_default_route_table.go]: Can we guarantee that the first RouteTable in the collection is the correct one?

Comment (stack72) [inline builtin/providers/aws/resource_aws_default_route_table.go]: I was being stupid here... comment revoked ;)

File: website/source/docs/providers/aws/r/vpc.html.markdown [hunk 4]
```diff
@@ -61,6 +61,7 @@ The following attributes are exported:
      [`aws_main_route_table_association`](/docs/providers/aws/r/main_route_table_assoc.html).
 * `default_network_acl_id` - The ID of the network ACL created by default on VPC creation
 * `default_security_group_id` - The ID of the security group created by default on VPC creation
+* `default_security_group_id` - The ID of the route table created by default on VPC creation
```
Comment (stack72) [inline website/source/docs/providers/aws/r/vpc.html.markdown]: this should be `default_route_table_id`

**Comment by stack72** [comment]
Few small things but this is really good - loved the note about being wary! I can forsee people getting themselves into trouble here so it's good to point this out in advance

File: builtin/providers/aws/resource_aws_default_route_table.go (hunk 1 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_default_route_table.go]: We want to revoke the rules regardless if there is any update to be made. The procedure is that during "creation" we wipe the slate clean. This is documented to warn people .

File: builtin/providers/aws/resource_aws_default_route_table.go (hunk 2 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_default_route_table.go]: on read we need to find the default route table for this VPC. Unlike Default Network ACLs or Default Security Groups, the Default Route Table can be changed (you can make a new one and then make that "default"). So we need to find if we are the default, and then refresh and show a diff (if there is one) accordingly.

File: builtin/providers/aws/resource_aws_default_route_table.go (hunk 3 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_default_route_table.go]: eh

File: website/source/docs/providers/aws/r/vpc.html.markdown (hunk 4 above)
Comment (catsby) [inline website/source/docs/providers/aws/r/vpc.html.markdown]: penguin

File: builtin/providers/aws/resource_aws_default_route_table.go (hunk 1 above)
Comment (stack72) [inline builtin/providers/aws/resource_aws_default_route_table.go]: 👍

**Comment by stack72** [comment]
The PR looks good to me - the questions were based more on my misunderstanding of how it would work. Explanations and the documentation help to clear this up

LGTM :)

**Pull request merged**